src/Redux/reducers/countries.js

```diff
@@ -0,0 +1,44 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+const url = 'https://restcountries.com/v3.1/all';
+
+export const fetchCountries = createAsyncThunk(
+  'FETCH',
+  async () => {
+    const data = await (fetch(url)).then((res) => res.json());
+    const response = data.map((country) => ({
+      name: country.name.common,
+      countrycode: country.cca2,
+      region: country.region,
+      flag: country.flags.svg,
+      capital: country.capital,
+    }));
+    return response;
+  },
+);
+
+export const filteredCountries = (data = '') => (
+  {
+    type: 'FILTER',
+    payload: data,
+  }
+);
+
+const countryReducer = (state = [], action) => {
+  switch (action.type) {
+    case 'FETCH/fulfilled': {
```

Comment:
- [**Optional**] Hi @Sajeelzafar, so far, all your action types have been strings. Though your application still works, however, in larger applications, using strings for action types can make it harder to track bugs :bug:  We can avoid these types of errors in the future by saving our action types using constants.

Reply:
Thank you @FrederickMih for your suggestion. I have renamed every action to constant and will try adopting it in the future as well.